### DISCUSSION TITLE
How to render many markers in the most performant way

### DISCUSSION THREAD

**Discussion by LamaAgent** [fleaflet/flutter_map#1497] [Q&A] [answered]
### What do you want implemented?

Hello people
I am currently programming an app that displays a map for an entire state.
I have a few thousand markers that I need to display. But to improve the performance, I would need now possibilities, for the following solutions:

- Render only the markers that are in sight
- Markers only from a zoom level, so that you can see individual cities/villages.

How can I render so many markers in the most performant way?

### Platforms Affected

Android, iOS

### Severity

Obtrusive: No workarounds are available, and this is essential to me

### Requirements

- [X] I agree to follow this project's [Code of Conduct](https://github.com/fleaflet/flutter_map/blob/master/CODE_OF_CONDUCT.md)
- [X] I am using the [latest stable version](https://pub.dev/packages/flutter_map) of this package
- [X] I have checked for similar feature requests which may be duplicates

**Comment by JaffaKetchup** [comment, marked as answer]
Hi there,
Markers are culled automatically - the are not rendered when off screen.
In terms of displaying at a certain zoom level, the `MapController` has a stream that can be listened to for events, such as changes in zooms. You can use this to control whether markers appear or not. You should filter the stream to include only events that you need to worry about, to avoid unnecessary state changes.
Also see https://docs.fleaflet.dev/plugins/list#marker-clustering.

**Reply by LamaAgent** [reply]
So it would be possible, for example, for the markers to be displayed from zoom level 13?  How can I hide/show the markers when I have a normal marker layer?

**Comment by ibrierley** [comment]
Few different options spring to mind, clustering may make sense, eg https://github.com/rorystephenson/flutter_map_supercluster   https://pub.dev/packages/flutter_map_marker_cluster . You could have a look at my alpha geojson thing at https://github.com/ibrierley/geojson_vector_slicer if the markers don't change (not so good for dynamic markers).

As @JaffaKetchup says, you could listen to a stream and display a different marker layer when on a different zoom level.

**Reply by LamaAgent** [reply]
Thanks for your answer. How can I display another (empty in my case) MarkerLayer from a certain zoom level?

**Reply by ibrierley** [reply]
Listen to the mapController stream, and if zoom == whatever, markerLayer = []; Take a peek at the map_controller example in the examples folder for how to listen to a stream etc.
